### DISCUSSION TITLE
FromFormAttribute is in wrong namespace.

### DISCUSSION THREAD

**Discussion by lbgitjp** [dotnet/aspnetcore#56162] [General] [answered]
```
var builder = WebApplication.CreateSlimBuilder(args);
var app = builder.Build();

app.MapPost("/UploadFiles", async ([FromForm] IFormFile myFile) =>
{
    string tempfile = Path.GetTempFileName();
    await using var stream = File.OpenWrite(tempfile);
    await myFile.CopyToAsync(stream);
});

app.Run();
```

in above code, if i use [FromForm], i need add below namespace:

`using Microsoft.AspNetCore.Mvc;`

but i don't use any Mvc features,  so the class Microsoft.AspNetCore.Mvc.FromFormAttribute( [FromForm]) is in wrong namespace, 

thanks!

**Comment by martincostello** [comment, marked as answer]
This is an historical artifact - the team are aware of it, but moving it would be a hugely breaking change.

If you wish to avoid using the attribute entirely because of its presence in the MVC namespace, then you can implement your own using `IFromFormMetadata` [like this](https://github.com/dotnet/aspnetcore/blob/7f2dfb0020ff29ab57df31e910f2b2d5ef48bf82/src/Http/Routing/test/FunctionalTests/MinimalFormTests.cs#L766-L770) which should work the same in a Minimal API.

**Reply by lbgitjp** [reply]
I see, thanks!
